Update docs for S22 (#94)
UMM-CSci-3601/intro-to-git: PART_2_JAVA_INTRODUCTIONS.md

> of the Black Lives Matter movement has been an increasing recognition of
> the painful connotations that term has for many people, and there have
> been [calls to change from `master` to the more neutral `main`](https://dev.to/afrodevgirl/replacing-master-with-main-in-github-2fjf).
> GitHub is in fact working to change the default naming for new projects,
> GitHub in fact did a lot of work to support changing the default naming for new projects,

> > and provide tools to help automate changing the name of the default branch.

Reviewer: quick verb tense fix

```suggestion
> and provides tools to help automate changing the name of the default branch.
```

> details. New, blank repositories will, for example, now
> have `main` as their default branch name, and they've provided tools for
> renaming your primary branch.
>
> There were a _lot_ of instances of `master` created when that was the
> default, so there are likely to be references to `master` as the default
> branch in older
> `git` examples and documentation for quite a while.

To illustrate the use of branches, we'll have each group create a new branch
for their modification of the shared program. Assuming, for example, that
we still have Pat and Chris, they would create a new branch called something
like `pat-and-chris-greetings`.

In GitKraken we can do this by clicking the `Branch` button in the toolbar
up at the top. It
won't immediately look like anything actually happened, but there will be a
text box to the left of the history diagram where the _HEAD_ (top) of `main` is.
Enter your branch name (e.g., `pat-and-chris-greetings`) there and hit return. It

> should now show (part of) :mask: `pat-and-chris-greetings` and a `+1` which indicates

Reviewer: ```suggestion
should now show (part of) `pat-and-chris-greetings` and a `+1` which indicates
```

that there's another branch (`main`) whose HEAD is at the same place. You should

> also have :mask: `pat-and-chris-greetings` listed as a new branch under `LOCAL` in the

Reviewer: ```suggestion
also have `pat-and-chris-greetings` listed as a new branch under `LOCAL` in the
```

makes sense to have them in the same order as the calls. Our tests don't check
that, though. (They probably can't actually, at least not in any easy way.)

> Once everything looks good, compile and run your program to make sure that everything is good; `./gradlew run` should do the trick.

Reviewer: Just a small phrasing suggestion (repeated words)

```suggestion
Once everything looks good, compile and run your program to make sure that it works correctly; `./gradlew run` should do the trick.
```

### Commit with GitKraken

To commit in GitKraken click the top line in the history graph; its circle
should be empty with a dotted line, and it should have `// WIP` where the
commit summary is.

> ![GitKraken history graph showing that there's work to commit](docs/images/GitKraken-wip.png)

Author: I tried to add screen grabs in various places to illustrate what was going on.

safe to just `push` and see what happens).

So hit `Push` in the GitKraken toolbar. It won't look like anything happened,
but there will be a question where the toolbar used to be:

![GitKraken asking what remote branch to use](docs/images/GitKraken-what-remote-branch.png)

The issue here is we created a new _local_ branch, and GitKraken wants to know
what _remote_ repository to connect that to, and with what name. It will default
to `origin` (which in this case is another name for the GitHub repository we

> cloned from) and our branch name (e.g., :mask: `pat-and-chris-greetings`). Those are

Reviewer: ```suggestion
cloned from) and our branch name (e.g., `pat-and-chris-greetings`). Those are
```
